content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: blog
+title: "Kubernetes 1.28: Persistent Volume Last Phase Transition Time"
+date: 2023-08-15T10:00:00-08:00
+slug: kubernetes-1-28-persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we are introducing a new alpha feature that aims to improve PersistentVolume (PV) management and help cluster administrators gain better insights into the lifecycle of PVs.
+With the addition of the LastPhaseTransitionTime field in the PV status, administrators will be able to track the last time a PV transitioned to a different phase, allowing for more efficient and informed resource management.
+
+## Why do we need new PV field
+
+PersistentVolumes in Kubernetes play a crucial role in providing storage resources to workloads running in the cluster.
+However, managing these PVs effectively can be challenging, especially when it comes to determining the last time a PV transitioned between different phases, such as "Pending," "Bound," or "Released."
+Administrators often need to know when a PV was last used or transitioned to certain phases, for instance, to implement retention policies, perform cleanup, or monitor storage health.
+
+In the past, Kubernetes users have faced data loss issues when using the "Delete" retain policy and had to resort to the safer "Retain" policy.
+By introducing the LastPhaseTransitionTime field, we aim to provide a more generic solution that can be used for various use cases,
+including manual cleanup based on the time a volume was last used or producing alerts based on phase transition times.
+
+## How LastPhaseTransitionTime helps
+
+The new LastPhaseTransitionTime field added to the PV status is updated every time a PV transitions from one phase to another.
+Whether it's transitioning from "Pending" to "Bound," "Bound" to "Released," or any other phase transition, the LastPhaseTransitionTime will be recorded.
+For newly created PVs the phase will be set to "Pending" and the LastPhaseTransitionTime will be recorded as well.
+
+This feature allows cluster administrators to:
+
+1. Implement Retention Policies
+
+   With the LastPhaseTransitionTime, administrators can now track when a PV was last used or transitioned to the "Released" phase.
+   This information can be crucial for implementing retention policies to clean up resources that have been in the "Released" phase for a specific duration.
+
+2. Monitor Storage Health
+
+   By analyzing the phase transition times of PVs, administrators can monitor storage health more effectively.
+   For example, they can identify PVs that have been in the "Pending" phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
+
+3. Produce Alerts and Metrics
+
+   The LastPhaseTransitionTime can be utilized to produce alerts and metrics.
```

Comment:
If we need such metrics, we need code in KCM / PV controller that would emit them. I'm not sure it's a goal of this feature.